engine/src/resource/json_utils.cpp

```diff
@@ -22,67 +119,136 @@
  * along with Vega Strike. If not, see <https://www.gnu.org/licenses/>.
  */
 
-#include "json_utils.h"
-
 #include <string>
 #include <vector>
+#include <stdexcept>
+
 #include <boost/algorithm/string.hpp>
 #include <boost/json.hpp>
 
+/**
+ * @brief Retrieve a nested JSON value, given a pipe-delimited path.
+ *        If no key is found at any step, returns an empty JSON value.
+ * @param object   The root JSON object.
+ * @param key      A string containing one or more subkeys, delimited by '|'.
+ * @return boost::json::value   The found value, or an empty value on failure.
+ */
+static boost::json::value JsonGetValue(const boost::json::object& object,
+                                       const std::string& key)
+{
+    std::vector<std::string> key_sections;
+    boost::split(key_sections, key, boost::is_any_of("|"));
 
+    const boost::json::object* current_obj = &object;
+    for (size_t i = 0; i < key_sections.size(); ++i) {
```

Comment:
How is this cleaner than 
`for(const std::string& key_section : key_sections)` ?